homeassistant/components/touchlinesl/climate.py

```diff
@@ -0,0 +1,111 @@
+"""Roth Touchline SL climate integration implementation for Home Assistant."""
+
+from typing import Any
+
+from pytouchlinesl import Zone
+from pytouchlinesl.client.models import GlobalScheduleModel
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TouchlineSLConfigEntry
+from .coordinator import TouchlineSLModuleCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TouchlineSLConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Touchline devices."""
+    account = entry.runtime_data
+
+    for module in await account.modules():
+        coordinator = TouchlineSLModuleCoordinator(hass, module=module)
+        await coordinator.async_config_entry_first_refresh()
+        async_add_entities(
+            (
+                TouchlineSLZone(coordinator=coordinator, zone_id=z)
+                for z in coordinator.data["zones"]
+            ),
+            True,
+        )
+
+
+CONST_TEMP_PRESET_NAME = "Constant Temperature"
+
+
+class TouchlineSLZone(CoordinatorEntity, ClimateEntity):
```

Comment:
New integrations are required to have `_attr_has_entity_name = True`. I will save you the details on this, but you also want to add a `_attr_device_info` in the constructor, where you will set the device name, and any identifiers you have. This way the integration will show devices. Once you have done that, you want to add `_attr_name = None`. This way, the name of the entity will follow the name of the device.

Reply:
Also, since only the zone has entities, should we manually add a parent device for the module? You can then reference the parent device in the zone devices via `via_device` in the device info. This way, you will create a module page, where you can see all linked zones like in the philips hue bridge:
![image](https://github.com/user-attachments/assets/b2bb67f8-3153-4e0a-a392-315c64cb915a)

Reply:
For manual device creation, check either `airgradient` or `ao_smith`

Reply:
Done, thanks!